src/class/SimpleData.ts

```diff
@@ -66,378 +87,425 @@ export default class SimpleData {
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    addBins(...args: any[]) {
-        const data = this.#apply(
-            addBins_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    addQuantiles({ 
+        key, 
+        newKey, 
+        nbQuantiles, 
+        overwrite = true 
+    }: { 
+        key: string, 
+        newKey: string, 
+        nbQuantiles: number, 
+        overwrite?: boolean 
+    }): SimpleData {
+        const data = addQuantiles_(this._data, key, newKey, nbQuantiles, this._verbose)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    addOutliers(...args: any[]) {
-        const data = this.#apply(
-            addOutliers_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    addBins({ 
+        key, 
+        newKey, 
+        nbBins, 
+        overwrite = true 
+    }: { 
+        key: string, 
+        newKey: string, 
+        nbBins: number, 
+        overwrite?: boolean 
+    }): SimpleData {
+        const data = addBins_(this._data, key, newKey, nbBins, this._verbose)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    excludeOutliers(...args: any[]) {
-        const data = this.#apply(
-            excludeOutliers_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    addOutliers({ 
+        key, 
+        newKey, 
+        overwrite = true 
+    }: { 
+        key: string, 
+        newKey: string, 
+        overwrite?: boolean 
+    }): SimpleData {
+        const data = addOutliers_(this._data, key, newKey, this._verbose)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    correlation(key1: string, key2: string, options: Options) {
-        // We deal with the parameters manually to deal with optional arguments
-        const data = this.#apply(
-            correlation_,
-            key1,
-            key2,
-            options === undefined ? this._defaultOptions : options
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    excludeOutliers({ key, overwrite = true }: { key: string, overwrite?: boolean }): SimpleData {
+        const data = excludeOutliers_(this._data, key, this._verbose)
+        overwrite && this.#updateSimpleData(data)
+
         return this
     }
 
-    addItems(...args: any[]) {
-        const data = this.#apply(
-            addItems_,
-            ...args
-        )
-        this.#updateSimpleData(data)
+    @logCall()
+    addItems({ 
+        dataToBeAdded, 
+        nbDigits = 1, 
+        overwrite = true 
+    }: { 
+        dataToBeAdded: SimpleDataItem[], 
+        nbDigits?: number, 
+        overwrite?: boolean 
+    }): SimpleData {
+        const data = addItems_(
+            this._data,
+            dataToBeAdded,
+            this._verbose,
+            nbDigits
```

Comment:
I think we could hardcode nbDigits in the function since it's just for a log. Otherwise, it's just some noise. And 1 would be enough.